rfc-102-add-notification-workflow.md

```diff
@@ -0,0 +1,26 @@
+# A 'notify' workflow step in publishing pipeline
+
+## Summary
+
+Email notifications are currently sent to relevant subscribers only when content is published with a major update.
+
+We've identified a use case where the significance of _organising_ content after it has been published may warrant an email notification.
+
+## Problem
+
+We have a high profile use case with the [business readiness finder](https://www.gov.uk/find-eu-exit-guidance-business) where the facet-tagging workflow step is deemed significant enough to warrant notifying subscribers via an email alert.
+
+Departments wish to notify users that new items of content from across GOV.UK have been included in the results for this finder. This is partly a consequence of tagging content after it has been published, but also in part due to the process of writing and reorganising guidance at pace.
+
+Republishing content for the purposes of notifying users is possible but this seems pollutive given the workflow event is (re)categorisation.
+
+The current (temporary) [tagging notification mechanism uses a direct HTTP request between Rummager and Email Alert API](https://github.com/alphagov/rummager/blob/master/lib/indexer/workers/metadata_tagger_notification_worker.rb#L11-L20).
+
+The current (temporary) [metadata tagging implementation is moving to Content Tagger](https://github.com/alphagov/content-tagger/pull/884), the publishing pipeline will handle [tagging via links](https://github.com/alphagov/govuk-content-schemas/commit/071731cb08e8c9c8956a70769748c979f762cf44).
+The current email tagging notifications will not be triggered once this has moved.
+
+## Proposal
+
+- Publishing API supports a 'notify' endpoint and sends a distinct message type downstream for Email Alert Service/API consumption.
+- Content Tagger UI supports an explicit 'notify users of these changes' workflow choice, defaulting to a no-op.
```

Comment:
It would be interested to know if you've considered having Content Tagger do what Rummager is doing now, and talking to the Email Alert API directly.

Or to put it another way, what's the benefit of involving the Publishing API and RabbitMQ.

Reply:
I did consider this, it felt out of place as it was another bypass of the publishing pipeline to get to a downstream API.
The current tagging email mechanism was a very quick hack in Rummager, I think there are sufficient benefits of predictability and visibility in using the publishing pipeline.

Reply:
@bilbof I've considered/discussed just using a major change as a way to send an additional notification to subscribers. It's quite pollutive of Edition change history though, each major change is recorded and in many formats it's also rendered. It feels like an awkward abuse of change history in order to generate a notification.
Ideally in the future facet tagging will happen as part of the draft-to-published workflow and this endpoint may well be deprecated and dropped. Some discussions I've had with developers identified a need to defer notifications though, for instance when dependency resolution is slow to update all related items, it may be desirable to have greater control over when a notification goes out. This RFC doesn't fully address this as we are working with the EU Exit finder tagging use case but it's an example of how this endpoint might be evolved.

Reply:
> I did consider this, it felt out of place as it was another bypass of the publishing pipeline to get to a downstream API.

I guess to me, talking to the Email Alert API directly seems a bit neater, as it involves less services, and I don't see what would be bypassed by not going through the Publishing API. I've only skimmed the relevant Publishing API Pull Request, but the only extra stateful change that occurs is the event in the events table right?

> The current tagging email mechanism was a very quick hack in Rummager, I think there are sufficient benefits of predictability and visibility in using the publishing pipeline.

As I mentioned above, the entry in the events table adds to the audit trail at least, which might be what you meant by visibility.

Anyway, adding support in to the Publishing API to talk indirectly to the Email Alert API doesn't preclude talking directly to the Email Alert API now, or in the future, so this is not a crucial point of discussion.

Reply:
@cbaines I'll go with consensus here. I don't particularly mind talking directly to the email alert api if that's agreeable to the tech community, thanks for the helpful info and discussion.